examples/add_pyramid.py

```diff
@@ -22,4 +22,4 @@
     viewer = napari.Viewer()
 
     # add image pyramid
-    viewer.add_pyramid(pyramid, clim_range=[0, 255])
+    viewer.add_pyramid(pyramid, contrast_limits_range=[0, 255])
```

Comment:
@jni I dislike have both the contrast_limits and contrast_limits_range as keyword arguments - I think it will be confusing to new commers, so figuring a way to get rid of the range option sounds great.

Are you proposing that instead of using min / max we take the dtype range? So (0, 255) for uint8, (-255, 255) for int8, (0, 2^16-1) for uint16 etc and then 0-1 for floats? I just worry there will be too many times this does something weird, and that taking the min / max of the image right now is nicer.

The problem is just that for lazy loading of large arrays the min/max is expensive to compute. We had just the min/max of the first displayed frame at one point, but that can also be confusing.

Also what should the behavior be if someone does pass a contrasts_limits keyword arg? One option would just be to use these values as the range of the sliders too, but then you couldn’t make the clims more extreme ever. Another option would be to set them at the 30% and 70% point of the slider, possibly with the exception of locking 0 to 0. Thoughts?

Reply:
@jni I dislike have both the contrast_limits and contrast_limits_range as keyword arguments - I think it will be confusing to new commers, so figuring a way to get rid of the range option sounds great.

Are you proposing that instead of using min / max we take the dtype range? So (0, 255) for uint8, (-255, 255) for int8, (0, 2^16-1) for uint16 etc and then 0-1 for floats? I just worry there will be too many times this does something weird, and that taking the min / max of the image right now is nicer.

The problem is just that for lazy loading of large arrays the min/max is expensive to compute. We had just the min/max of the first displayed frame at one point, but that can also be confusing.

Also what should the behavior be if someone does pass a contrasts_limits keyword arg? One option would just be to use these values as the range of the sliders too, but then you couldn’t make the clims more extreme ever. Another option would be to set them at the 30% and 70% point of the slider, possibly with the exception of locking 0 to 0. Thoughts?

Reply:
@sofroniewn my proposal is to have a single `contrast_limits` keyword argument set to None by default. If it is set to anything else, then *that is the min and max*, and the sliders are at those positions to begin with. The user needs to move them manually or set them in the next line. It would be unsupported to set both `contrast_limits_range` and `contrast_limits` in that same line.

Does this make sense?

Incidentally, I do think we should deprecate using the min/max and use instead the data ranges and instead use the scikit-image ranges https://scikit-image.org/docs/dev/user_guide/data_types.html, with a way to override them manually. It's too easy to add a dask array and have the computation take forever, with no feedback to the user.

Reply:
ok sounds reasonable - i will drop the `contrast_limits_range` keyword arg from this PR and make the default behavior `min/max` - we can decide about default behaviour later

you'll still be able to change `contrast_limits_range` by directly setting the property